### ISSUE TITLE
Compile error under OpenBSD opendmarc.c:1482: error: invalid application of 'sizeof' to incomplete type 'struct hsearch_data'

### ISSUE THREAD

**Issue by thegushi** [trusteddomainproject/OpenDMARC#130]
This is a confounding error: Only happens under GCC.  (Note that the OpenBSD compile is still blocked by the hsearch thing), but the openBSD reporter has been using clang to compile.

```
gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -I/usr/local/include -I./../libopendmarc -DCONFIG_BASE=\"/usr/local/etc\" -pthread -g -O2 -MT opendmarc-config.o -MD -MP -MF .deps/opendmarc-config.Tpo -c -o opendmarc-config.o `test -f 'config.c' || echo './'`config.c
mv -f .deps/opendmarc-config.Tpo .deps/opendmarc-config.Po
gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -I/usr/local/include -I./../libopendmarc -DCONFIG_BASE=\"/usr/local/etc\" -pthread -g -O2 -MT opendmarc-opendmarc.o -MD -MP -MF .deps/opendmarc-opendmarc.Tpo -c -o opendmarc-opendmarc.o `test -f 'opendmarc.c' || echo './'`opendmarc.c
opendmarc.c: In function 'dmarcf_config_load':
opendmarc.c:1482: error: invalid application of 'sizeof' to incomplete type 'struct hsearch_data'
opendmarc.c:1483: warning: implicit declaration of function 'hcreate_r'
opendmarc.c:1537: warning: implicit declaration of function 'hsearch_r'
opendmarc.c: In function 'mlfi_eom':
opendmarc.c:2401: warning: implicit declaration of function 'dmarcf_mail_parse_multi'
opendmarc.c: In function 'dmarcf_config_free':
opendmarc.c:4065: warning: implicit declaration of function 'hdestroy_r'
*** Error 1 in opendmarc (Makefile:594 'opendmarc-opendmarc.o')
*** Error 1 in opendmarc (Makefile:864 'all-recursive': @fail=;  if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_ru...)
*** Error 1 in . (Makefile:510 'all-recursive': @fail=;  if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_running_wi...)
*** Error 2 in /root/opendmarc (Makefile:396 'all')
```

**Comment by thegushi** [comment]
See my comments under #94 and pull #57 -- I think it's the wrong idea for *us* to ship this library, but I'm looking to find what the best-considered *sane* way to tell other people (such as whomever will maintain this port for OpenBSD) to do it.  I'm blocked on response from those folks.  (Github issues don't have a "stalled" status, annoyingly)

**Comment by thegushi** [comment]
Per a quick conversation, it looks like the right answer is as you suggest, and @mskucherawy has said he'll take a stab at it.  The function is only used for a whitelist, and is read-only so we're not super concerned about the performance here, thus not needing the reentrant values.  (Paraphrased)

**Comment by bicknell** [comment]
The reentrant ones should have no performance difference. 

The non-reentrant can only manage _one_ hash table global _per binary program_.  The reentrant ones can manage _multiple_ hash tables by moving the data structure for the hash to outside of the functions.  E.g. allowing two threads to have their own hash tables.

I'm unclear if the code here ever only creates one, or if there is a case where it creates multiple.  I have to wonder if it creates multiple since someone went to the trouble of using it.

**Comment by thegushi** [comment]
Okay.  Replacing hcreate_r() with hcreate() will solve this under OpenBSD and MacOS.  I'm going to create a new issue just for this, so it's not under the big meta "All the openbsd things" ticket and is an easy to accomplish bit, and close this one out, since if that warning is still present after that, it means my os is broken. :)
